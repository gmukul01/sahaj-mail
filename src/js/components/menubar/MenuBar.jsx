import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Folders from './Folders';
import Labels from './Labels';
import Categories from './Categories';
import ComposeEmail from 'components/email/ComposeEmail';

const MenuBar = ({ totalUnread, sendEmail }) => {
	const [showModal, setModalState] = useState(false),
		onCloseModal = () => setModalState(false),
		onOpenModal = () => setModalState(true);

	return (
		<section className="menubar">
			<button className="button-solid menubar-compose" onClick={onOpenModal}>
				Compose Mail
			</button>
			<ComposeEmail {...{ showModal, onCloseModal, sendEmail }} />
			<Folders totalUnread={totalUnread} />
			<Categories />
			<Labels />
		</section>
	);
};

MenuBar.propTyoes = {
	totalUnread: PropTypes.number,
	sendEmail: PropTypes.func
};

export default React.memo(MenuBar);
