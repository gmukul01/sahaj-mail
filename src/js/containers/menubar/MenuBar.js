import { connect } from 'react-redux';

import { sendEmail } from 'actions/emails';
import MenuBar from 'components/menubar/MenuBar';

const mapStateToProps = ({ user: { name, email }, inbox: { pageNumber, emailsPerPage, totalUnread } }) => {
	return {
		name,
		email,
		pageNumber,
		emailsPerPage,
		totalUnread
	};
};

const mergeProps = (stateProps, dispatchProps, ownProps) => {
	const { sendEmail } = dispatchProps;
	const { name, email, pageNumber, emailsPerPage } = stateProps;
	return {
		...stateProps,
		...dispatchProps,
		...ownProps,
		sendEmail(emailDetais) {
			const newEmailDetails = {
				...emailDetais,
				from: { name, email },
				isRead: false,
				hasAttachment: false
			};
			sendEmail(newEmailDetails, 'inbox', pageNumber, emailsPerPage);
		}
	};
};

export default connect(
	mapStateToProps,
	{
		sendEmail
	},
	mergeProps
)(MenuBar);
