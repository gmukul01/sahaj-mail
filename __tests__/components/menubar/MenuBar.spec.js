import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import MenuBar from 'components/menubar/MenuBar';

describe('MenuBar Component', () => {
	const mockSendEmail = jest.fn(),
		initialProps = {
			totalUnread: 2,
			sendEmail: mockSendEmail
		};

	it('should match snapshot', () => {
		const menubar = renderer
			.create(
				<BrowserRouter>
					<MenuBar {...initialProps} />
				</BrowserRouter>
			)
			.toJSON();
		expect(menubar).toMatchSnapshot();
	});

	it('should call mockSendEmail when sendEmail is called', () => {
		const menubar = shallow(<MenuBar {...initialProps} />);
		menubar.find('ComposeEmail').prop('sendEmail')('Dummy');
		expect(mockSendEmail.mock.calls[0][0]).toBe('Dummy');
	});
});
