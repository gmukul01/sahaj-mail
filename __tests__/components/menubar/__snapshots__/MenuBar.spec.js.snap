// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar Component should match snapshot 1`] = `
<section
  className="menubar"
>
  <button
    className="button-solid menubar-compose"
    onClick={[Function]}
  >
    Compose Mail
  </button>
  <div
    className="modal"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h3
          className="title"
        >
          New Message
        </h3>
        <button
          className="modal-close"
          onClick={[Function]}
        >
          Ã—
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          className="compose-email"
          onSubmit={[Function]}
        >
          <label>
            To
          </label>
          <input
            onChange={[Function]}
            placeholder="Enter recipient email id"
            type="text"
            value=""
          />
          <label>
            Subject
          </label>
          <input
            onChange={[Function]}
            placeholder="Enter subject"
            type="text"
            value=""
          />
          <label>
            Body
          </label>
          <textarea
            onChange={[Function]}
            placeholder="Enter message"
            value=""
          />
          <button
            className="button-solid"
            type="submit"
          >
            Send
          </button>
        </form>
        
      </div>
    </div>
  </div>
  <div
    className="menubar-folders"
  >
    <h5>
      FOLDERS
    </h5>
    <ul>
      <li>
        <span>
          <i
            className="fas fa-inbox"
          />
          Inbox
        </span>
        <button
          className="menubar-folders-inbox chips-btn"
        >
          2
        </button>
      </li>
      <li>
        <span>
          <i
            className="fas fa-envelope"
          />
          Send Email
        </span>
      </li>
      <li>
        <span>
          <i
            className="fas fa-certificate"
          />
          Important
        </span>
      </li>
      <li>
        <span>
          <i
            className="fas fa-file-alt"
          />
          Drafts
        </span>
        <button
          className="menubar-folders-drafts chips-btn"
        >
          2
        </button>
      </li>
      <li>
        <span>
          <i
            className="fas fa-trash-alt"
          />
          Trash
        </span>
      </li>
    </ul>
  </div>
  <div
    className="menubar-categories"
  >
    <h5>
      CATEGORIES
    </h5>
    <ul>
      <li
        className="menubar-categories-work"
      >
        <i
          className="fas fa-circle"
        />
        Work
      </li>
      <li
        className="menubar-categories-documents"
      >
        <i
          className="fas fa-circle"
        />
        Documents
      </li>
      <li
        className="menubar-categories-social"
      >
        <i
          className="fas fa-circle"
        />
        Social
      </li>
      <li
        className="menubar-categories-advertising"
      >
        <i
          className="fas fa-circle"
        />
        Advertising
      </li>
      <li
        className="menubar-categories-clients"
      >
        <i
          className="fas fa-circle"
        />
        Clients
      </li>
    </ul>
  </div>
  <div
    className="menubar-labels"
  >
    <h5>
      LABELS
    </h5>
    <div>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Film
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Photography
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Music
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Holidays
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Children
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Home
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Work
      </button>
      <button
        className="button-outlined"
      >
        <i
          className="fas fa-tag"
        />
        Family
      </button>
    </div>
  </div>
</section>
`;
